### PR TITLE
Adding service support to logstash

### PR DESCRIPTION
**Pull request by rickmark**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/logstash.rb [hunk 1]
```diff
@@ -39,6 +39,37 @@ def caveats; <<-EOS.undent
...
+          <array>
+            <string>#{HOMEBREW_PREFIX}/bin/logstash</string>
+          </array>
+          <key>EnvironmentVariables</key>
+          <dict>
+          </dict>
+          <key>RunAtLoad</key>
+          <true/>
+          <key>WorkingDirectory</key>
+          <string>#{var}</string>
```
Comment (MikeMcQuaid) [inline Formula/logstash.rb]: Stick this in `prefix` or similar so it doesn't start writing random files in `var` (which is shared between formulae)

File: Formula/logstash.rb [hunk 2]
```diff
@@ -39,6 +39,37 @@ def caveats; <<-EOS.undent
     EOS
   end
 
+  plist_options :manual => "logstash"
+
+  def plist; <<-EOS.undent
+      <?xml version="1.0" encoding="UTF-8"?>
```
Comment (MikeMcQuaid) [inline Formula/logstash.rb]: Can be undented by two characters

File: Formula/logstash.rb [hunk 3]
```diff
@@ -39,6 +39,37 @@ def caveats; <<-EOS.undent
...
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>KeepAlive</key>
+          <false/>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{HOMEBREW_PREFIX}/bin/logstash</string>
```
Comment (MikeMcQuaid) [inline Formula/logstash.rb]: Use `opt_bin` instead of `HOMEBREW_PREFIX`

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/logstash.rb [hunk 4]
```diff
@@ -39,6 +39,37 @@ def caveats; <<-EOS.undent
...
+        <array>
+          <string>#{opt_bin}/logstash</string>
+        </array>
+        <key>EnvironmentVariables</key>
+        <dict>
+        </dict>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}/logstash</string>
```
Comment (MikeMcQuaid) [inline Formula/logstash.rb]: I don't think this directory will exist by default.

Comment (dunn) [inline Formula/logstash.rb]: Ping on this comment

**Comment by fxcoudert** [comment]
Closing for now, please reopen once the comment is addressed.

**Pull request closed without merging**